examples/integrations/hf_trainer_api/det_callback.py

```diff
@@ -0,0 +1,348 @@
+
+            if round(state.epoch) >= self.current_op.length:
+                self._update_searcher(state, control)
+
+    def _update_searcher(self, state: TrainerState, control: TrainerControl) -> None:
+        if state.is_world_process_zero:
+            if self.last_metrics is None:
+                logging.warning(
+                    f"No training or evaluation metrics has been recorded. Please check your settings for "
+                    f"training metrics (--logging_strategy and --logging_steps) or "
+                    f"evaluation metrics (--evaluation_strategy and --eval_steps). "
+                    f"Reporting trainer_state.best_metric to the searcher."
+                )
+                self.current_op.report_completed(state.best_metric)
+            elif self.searcher_metric not in self.last_metrics:
+                logging.warning(
+                    f"Searcher metric {self.searcher_metric} from the yaml config file does not match any "
+                    f"of the recorded metrics in {self.last_metrics}. "
+                    f"Reporting trainer_state.best_metric to the searcher."
+                )
+                self.current_op.report_completed(state.best_metric)
+            else:
+                self.current_op.report_completed(
+                    self.last_metrics[self.searcher_metric]
+                )
+
+        try:
+            self.current_op = next(self.searcher_ops)
+        except StopIteration:
+            control.should_training_stop = True
+
+    def _check_searcher_compatibility(self, args: TrainingArguments) -> None:
+        if self.searcher_unit == "batches":
+            if args.max_steps == -1:
+                self._log_config_mismatch("epochs", args.num_train_epochs)
+            elif args.max_steps != self.searcher_max_length:
+                self._log_config_mismatch("batches", args.max_steps)
+        elif self.searcher_unit == "epochs":
+            if args.max_steps != -1:
+                self._log_config_mismatch("batches", args.max_steps)
+            elif args.num_train_epochs != self.searcher_max_length:
+                self._log_config_mismatch("epochs", args.num_train_epochs)
+
+    def _log_config_mismatch(
+        self,
+        trainer_units: str,
+        trainer_len: float,
+    ) -> None:
+        logging.warning(
+            f"Searcher configuration does not match HF Trainer configuration. "
+            f"Searcher uses {self.searcher_unit}={self.searcher_max_length}, "
+            f"while HF Trainer uses {trainer_units}={trainer_len}. "
+            f"Continuing this run may cause Searcher not to behave correctly. "
+            f"Make sure to match the units between HF Trainer and Searcher: "
+            f"use (--num_train_epochs and searcher.max_length.epochs) OR "
+            f"(--max_steps and searcher.max_length.batches)."
+        )
+
+
+def is_determined_available() -> bool:
+    return importlib.util.find_spec("determined") is not None
```

Comment:
why not just 
```python
try:
    import determined as det
    have_det = True
except:
    have_det = False
```
?

Reply:
No particular reason. I was following other Trainer callback integrations. Anyway, that will not be needed anymore after core_context is initialized in the main script.